mitiq/zne/inference.py

```diff
@@ -302,7 +304,9 @@ def _noise_to_expval(scale_factor, **exec_params) -> float:
 
         return self.iterate(_noise_to_expval, max_iterations)
 
-    def __eq__(self, other):
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, Factory):
+            return False
```

Comment:
Does it make sense to allow comparisons to things that aren't Factory type objects? Should it be
```suggestion
    def __eq__(self, other: Factory) -> bool:
        if not isinstance(other, Factory):
            return False
```

Reply:
This applies to a few other __eq__ methods below but I won't duplicate the comment

Reply:
That's a great question! It should in fact be `Any`, and if you change it to something else here's what happens:
```
mitiq/zne/inference.py:757: error: Argument 1 of "__eq__" is incompatible with supertype "object"; supertype defines the argument type as "object"
mitiq/zne/inference.py:757: note: This violates the Liskov substitution principle
mitiq/zne/inference.py:757: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
mitiq/zne/inference.py:757: note: It is recommended for "__eq__" to work with arbitrary objects, for example:
mitiq/zne/inference.py:757: note:     def __eq__(self, other: object) -> bool:
mitiq/zne/inference.py:757: note:         if not isinstance(other, ExpFactory):
mitiq/zne/inference.py:757: note:             return NotImplemented
mitiq/zne/inference.py:757: note:         return <logic to compare two ExpFactory instances>
```
The equality comparison method is defined by `object` to accept any type as `other` and thus it violates the [Liskov substitution principle](https://en.wikipedia.org/wiki/Liskov_substitution_principle) to change it (this is one of the tenets of good object-oriented SWE, part of a collection called [SOLID](https://en.wikipedia.org/wiki/SOLID)).